### ISSUE TITLE
Xiaomi 12x (psyche) Paranoid Android LHDC not working

### ISSUE THREAD

**Issue by Ilia-yash** [anonymix007/ExtA2DP#9]
On MIUI it originally supported LHDC, earbuds support LHDC as well. But when LHDC is enabled, independently of bitrate or other parameters it doesn't work, there's no sound at all. LDAC seems to work fine if bitrate is selected manually.
![Screenshot_20231017-215709_Magic](https://github.com/anonymix007/ExtA2DP/assets/54913721/6de07b3b-5cd6-4a6a-8fd5-000280d8f070)
![Screenshot_20231017-215602_Shell](https://github.com/anonymix007/ExtA2DP/assets/54913721/c83d5ce0-2f01-4e87-b7f7-e39f1d27c4dc)
![Screenshot_20231017-215621_Shell](https://github.com/anonymix007/ExtA2DP/assets/54913721/c1642bd1-a273-46de-ba4c-1e7d9bf3e5aa)
![Screenshot_20231017-215641_ExtA2DP](https://github.com/anonymix007/ExtA2DP/assets/54913721/7a51ca89-cbea-4013-a187-fd5cb7517339)

**Comment by anonymix007** [comment]
Provide logs.

**Comment by Ilia-yash** [comment]
I'm sorry for the dumb question, but what logs exactly? Where to get them? Here's the log folder from logs saved from LSposed https://dropmefiles.com/6nFLL

**Comment by anonymix007** [comment]
https://github.com/anonymix007/ExtA2DP#logs

**Comment by Ilia-yash** [comment]
Here: https://dropmefiles.com/6eylS

**Comment by anonymix007** [comment]
Another broken Xiaomi device. Maintainer will have to fix it somehow.
On my phone both `is_aidl_hal_available` and `is_qc_hal_available` are 1, you seem to have only qc hal. Maintainer will have to fix it somehow.

**Comment by anonymix007** [comment]
Can you run `sudo getprop persist.bluetooth.bluetooth_audio_hal.disabled` and post the output?

**Comment by anonymix007** [comment]
The issue *is* device-specific (more like vendor-specific). I haven't seen any instances of this happening on non-Xiaomi devices. It's probably related to the fact that Xiaomi did implement LHDC HW offload.
I have a feeling that just enabling AIDL will help.
Alternatively, you might try disabling A2DP Hardware Offload for now.

I don't know who's the maintainer, can they ping me in PA telegram group to discuss this?

**Comment by Ilia-yash** [comment]
Disabled hardware offload, now the music is playing on the device speakers 😅 
Previously there was no sound at all.
@mickaelmendes50 is Xiaomi 12X maintainer.

**Comment by anonymix007** [comment]
Run `dumpsys media.audio_policy | awk ' 
        /^ Config source: / {
            print $3
        }' ` and upload this file + name.

**Comment by Ilia-yash** [comment]
Here: https://dropmefiles.com/EO4dH
Path: /vendor/etc/audio/audio_policy_configuration.xml

**Comment by anonymix007** [comment]
Looks fine. Assuming you have Magisk companion module installed, it should be working almost fine (but switching between codecs needs to be done through LHDC, also aptX codecs will not work). 
Does LHDC present in module app? 
Does the line `APM::HwModule: createDevice: could not find HW module for device 0080 address ` present in logcat?

**Comment by Ilia-yash** [comment]
I have audio_policy_changer_magisk.zip installed, do I need some additional module?
LHDC is present in ExtA2DP
![Screenshot_20231018-214733_ExtA2DP](https://github.com/anonymix007/ExtA2DP/assets/54913721/d3735156-41a5-4791-b4b8-146454ea3cfb)

There's no such line in logcat.

But if Hardware offload is disabled, the sound comes from speakers, and if it's enabled, there's no sound at all, silence.

**Comment by anonymix007** [comment]
Have you tried LHDC V4? Currently you seem to be using LLAC, try changing bitrate to 500 or 900 for LHDC V4

**Comment by Ilia-yash** [comment]
Yeah, I have tried all bitrates and Sample Rates combinations, nothing helps.
